cmd/rokka/cli/massupload/scanner.go

```diff
@@ -0,0 +1,58 @@
+package massupload
+
+import (
+	"sort"
+	"path/filepath"
+	"os"
+	"fmt"
+	"strings"
+)
+
+func Scan(root string, images chan string, recursive bool, extensions []string) error {
+	// Sort extensions before scanning
+	sort.Strings(extensions)
+
+	defer close(images)
```

Comment:
isn't that closing the images chan too early? can't it be that you are still processing published images while this tries to close it already?